### ISSUE TITLE
Chunked responses

### ISSUE THREAD

**Issue by back2dos** [haxetink/tink_http#27]
This came up in #18.

To my understanding, this could be implemented as a function of type `OutgoingResponse->OutgoingResponse`. If the argument already has a `Content-Length` it would be returned as is, otherwise a new response would be constructed that uses chunked encoding. The tricky bit is in writing the `IdealSource->IdealSource` transformation for the body.

Not sure when I'll get around to do this. Volunteers are welcome. Or let me know if this becomes a pressing issue ;)

**Comment by kevinresol** [comment]
was trying to implement `OutgoingResponse.chunked()`. It turns out that nodejs uses chunked encoding by default, meaning that one doesn't need to set content-length nor transfer-encoding. Simply `write()` to the ServerResponse object and `end()` it, nodejs will take care of the else.

So if we transform the body into "chunked style" and send through nodejs, it will be shown "as-is" on the receiver side.

**Comment by back2dos** [comment]
> So I mean this should probably be handled at container level

Yes and no. First, there should be a standalone functionality that is of type `IncomingRequestHeader->OutgoingReponse->OutgoingResponse` that is able to fix a response so that makes sure the `content-length`, `transfer-encoding` and `connection` headers fit with one another and what the client supports (e.g. support for chunked responses must be announced by the client IIRC, otherwise connection: closed must be used).

**Comment by bablukid** [comment]
same problem here, tink_http ( PHP7 target ) throws the same error when acessing Google Geocoding API ( i.e "https://maps.googleapis.com/maps/api/geocode/json?address=Place%20de%20l%27%C3%A9toile%20Paris" )

**Comment by 0b1kn00b** [comment]
Wait, Why are both these issues closed?

**Comment by back2dos** [comment]
Which ones? ^^

**Comment by 0b1kn00b** [comment]
Regarding `SocketClient` not supporting chunked responses. Got bogged down with the Chunked implmentation thus far.

**Comment by kevinresol** [comment]
chunked encoding is not implemented in the SocketClient right now, and this issue is an umbrella covering that and all other chunk-related things.
